frame/support/src/lib.rs

```diff
@@ -1018,3 +1022,1000 @@ mod tests {
 		})
 	}
 }
+
+/// prelude to be used alongside pallet macro, for ease of use.
+pub mod pallet_prelude {
+	pub use sp_std::marker::PhantomData;
+	pub use frame_support::traits::{Get, Hooks, IsType, GetPalletVersion, EnsureOrigin};
+	#[cfg(feature = "std")]
+	pub use frame_support::traits::GenesisBuild;
+	pub use frame_support::dispatch::{DispatchResultWithPostInfo, Parameter, DispatchError};
+	pub use frame_support::storage::types::{
+		StorageValue, StorageMap, StorageDoubleMap, ValueQuery, OptionQuery,
+	};
+	pub use frame_support::{
+		EqNoBound, PartialEqNoBound, RuntimeDebugNoBound, DebugNoBound, CloneNoBound
+	};
+	pub use codec::{Encode, Decode};
+	pub use sp_inherents::{InherentData, InherentIdentifier, ProvideInherent};
+	pub use crate::weights::{DispatchClass, Pays, Weight};
+	pub use crate::{
+		Twox256, Twox128, Blake2_256, Blake2_128, Identity, Twox64Concat, Blake2_128Concat,
+		debug, ensure, RuntimeDebug, storage
+	};
+	pub use sp_runtime::{
+		traits::{MaybeSerializeDeserialize, Member, ValidateUnsigned},
+		transaction_validity::{
+			TransactionSource, TransactionValidity, ValidTransaction, TransactionPriority,
+			TransactionTag, TransactionLongevity, TransactionValidityError, InvalidTransaction,
+			UnknownTransaction,
+		},
+	};
+}
+
+/// `pallet` attribute macro allows to define a pallet to be used in `construct_runtime!`.
+///
+/// It is define by a module item:
+/// ```ignore
+/// #[pallet]
+/// mod pallet {
+/// ...
+/// }
+/// ```
+///
+/// Inside the module the macro will parse item with the attribute: `#[pallet::*]`, some attributes
+/// are mandatory, some other optional.
+///
+/// The attribute are explained with the syntax of non instantiable pallets, to see how pallet with
+/// instance work see below example.
+///
+/// Note various type can be automatically imported using pallet_prelude in frame_support and
+/// frame_system:
+/// ```ignore
+/// #[pallet]
+/// mod {
```

Comment:
does this work?
`mod {}` is invalid Rust AFAICT

Reply:
ah thanks it was a typo, it is now fixed in https://github.com/paritytech/substrate/pull/6877/commits/8e8ee8ddc60c1c740a14118147598a3098fad30d